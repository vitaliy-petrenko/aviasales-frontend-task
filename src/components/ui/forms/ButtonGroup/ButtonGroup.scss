@import '../../../../styles/abstract';

.button-group {
  display: flex;

  .btn {
    flex: 1;
    border-radius: 0;
    padding-left: $grid-xxs;
    padding-right: $grid-xxs;

    &:focus {
      z-index: 1;
    }

    &:first-child {
      border-radius: $border-radius-base 0 0 $border-radius-base;
    }

    &:last-child {
      border-radius: 0 $border-radius-base $border-radius-base 0;
    }
  }
}
