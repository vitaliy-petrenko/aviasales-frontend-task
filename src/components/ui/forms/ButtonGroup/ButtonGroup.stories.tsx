import React from 'react'
import ButtonGroup from './ButtonGroup'
import { storiesOf } from '@storybook/react'
import results from '../../../../../.jest-test-results.json'
import { withTests } from '@storybook/addon-jest'
import Button from '../Button'
import { action } from '@storybook/addon-actions'

const params = {
  jest: ['ButtonGroup.spec.tsx'],
}

storiesOf('ButtonGroup', module)
  .addDecorator(withTests({ results }))
  .add('ButtonGroup', () => (
      <ButtonGroup>
        <Button div onClick={action('clicked')}>Item</Button>
        <Button div onClick={action('clicked')} type='secondary'>Item</Button>
      </ButtonGroup>
    ), params
  )

