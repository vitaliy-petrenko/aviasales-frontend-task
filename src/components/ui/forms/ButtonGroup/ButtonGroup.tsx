import React from 'react'
import './ButtonGroup.scss'

const ButtonGroup: React.FC = ({ children }) => <div className='button-group'>{children}</div>

export default ButtonGroup
