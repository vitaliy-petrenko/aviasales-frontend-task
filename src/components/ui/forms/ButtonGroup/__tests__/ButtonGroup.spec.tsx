import React from 'react'
import ButtonGroup from '../index'
import { shallow } from 'enzyme'

describe('ButtonGroup component', () => {
  const component = shallow(<ButtonGroup/>)

  it('renders properly', () => {
    expect(component).toMatchSnapshot()
  })
})
