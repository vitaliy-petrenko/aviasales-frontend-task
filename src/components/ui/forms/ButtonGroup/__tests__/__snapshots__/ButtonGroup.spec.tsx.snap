// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup component renders properly 1`] = `
<div
  className="button-group"
/>
`;
